### PR TITLE
Improve error messages with interface name

### PR DESCRIPTION
**Pull request by oherrala**
Also provide From trait implementation for io::Error for most obvious error cases.

### CODE REVIEW

File: luomu-libpcap/src/error.rs [hunk 1]
```diff
@@ -53,17 +54,36 @@ impl fmt::Display for Error {
...
-            Error::OnlySupportedInMonitorMode => write!(f, "libpcap: The operation is supported only in monitor mode (PCAP_ERROR_NOT_RFMON)."),
-            Error::PermissionDenied => write!(f, "libpcap: The process doesn't have permission to open the capture source (PCAP_ERROR_PERM_DENIED)."),
-            Error::InterfaceNotUp => write!(f, "libpcap: The capture source device is not up (PCAP_ERROR_IFACE_NOT_UP)."),
-            Error::TimestampTypeNotSupported => write!(f, "libpcap: This device doesn't support setting the time stamp type (PCAP_ERROR_CANTSET_TSTAMP_TYPE)."),
-            Error::PromiscuousPermissionDenied => write!(f, "libpcap: The process has permission to open the capture source but doesn't have permission to put it into promiscuous mode (PCAP_ERROR_PROMISC_PERM_DENIED)."),
-            Error::TimestampPrecisionNotSupported => write!(f, "libcap: The requested time stamp precision is not supported (PCAP_ERROR_TSTAMP_PRECISION_NOTSUP)."),
-
+            Error::NotActivated(interface) => {
+                write!(f, "libpcap: The capture handle {} needs to be activated (PCAP_ERROR_NOT_ACTIVATED).", interface)
```
Comment (jtt) [inline luomu-libpcap/src/error.rs:58]: Change wording as we have name of the interface as parameter (maybe 'interface {} not opened" etc.)

File: luomu-libpcap/src/error.rs [hunk 2]
```diff
@@ -53,17 +54,36 @@ impl fmt::Display for Error {
...
+            }
+            Error::AlreadyActivated(interface) => {
+                write!(f, "libpcap: Capture handle {} already activated (PCAP_ERROR_ACTIVATED).", interface)
```
Comment (jtt) [inline luomu-libpcap/src/error.rs:61]: Here we also give name of the interface as parameter, but talk about handle in error message

File: luomu-libpcap/src/error.rs [hunk 3]
```diff
@@ -53,17 +54,36 @@ impl fmt::Display for Error {
...
+            }
+            Error::NoSuchDevice(interface) => {
+                write!(f, "libpcap: The capture source {} when the handle was created doesn't exist (PCAP_ERROR_NO_SUCH_DEVICE).", interface)
```
Comment (jtt) [inline luomu-libpcap/src/error.rs:64]: Maybe just "capture source {} does not exist", that would be much clearer.

File: luomu-libpcap/src/error.rs [hunk 4]
```diff
@@ -91,3 +111,33 @@ impl From<std::ffi::FromBytesWithNulError> for Error {
...
+                format!("could not open {}, permission denied", interface),
+            ),
+            Error::InterfaceNotUp(interface) => io::Error::new(
+                io::ErrorKind::Other,
+                format!("could not open {}, interface not up", interface),
+            ),
+            Error::PromiscuousPermissionDenied(interface) => io::Error::new(
+                io::ErrorKind::PermissionDenied,
+                format!(
+                    "could not set inteface {} in promiscous mode, permission denied",
```
Comment (jtt) [inline luomu-libpcap/src/error.rs:136]:
```suggestion
                    "could not set interface {} to promiscuous mode, permission denied",
```

**Review by jtt — CHANGES_REQUESTED** [review]

File: luomu-libpcap/src/error.rs (hunk 1 above)
Comment (oherrala) [reply luomu-libpcap/src/error.rs]: Now it is "Capture handle for interface {}..."

File: luomu-libpcap/src/error.rs (hunk 2 above)
Comment (oherrala) [reply luomu-libpcap/src/error.rs]: Now it is "Capture handle for interface {}"

File: luomu-libpcap/src/error.rs (hunk 3 above)
Comment (oherrala) [reply luomu-libpcap/src/error.rs]: "Capture interface {} doesn't exist"

**Review by jtt — APPROVED** [review]

**Pull request merged**